### PR TITLE
Signed responses and other small items

### PR DESCRIPTION
**Pull request by roshanr95**
- Responses are signed
- Tx hash is base32 now instead of hex to fit withing DNS limits
- Responses are timestamped

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,73 +11,38 @@ Oyster Serverless is a cutting-edge, high-performance serverless computing platf
...
+cargo build --release --target x86_64-unknown-linux-musl && sudo ./target/x86_64-unknown-linux-musl/release/serverlessrust
 ```
 
 <b>Make a request to the serveless application :</b>
 
 This transaction hash contains the JavaScript code that finds the prime factors of a given number :
 <a href="https://goerli.arbiscan.io/tx/0xc7d9122f583971d4801747ab24cf3e83984274b8d565349ed53a73e0a547d113">0xc7d9122f583971d4801747ab24cf3e83984274b8d565349ed53a73e0a547d113</a>
 
-Endpoint (POST) : `http://localhost:6000/api/serverless`
+Endpoint (POST) : `http://0xc7d9122f583971d4801747ab24cf3e83984274b8d565349ed53a73e0a547d113.localhost:6000/api/serverless`
```
Comment (vg-27) [inline README.md:32]: update example to base32 encoding

File: src/tests.rs [hunk 2]
```diff
@@ -341,4 +412,29 @@ pub mod serverlesstest {
             "worker timed out\n\nCaused by:\n    deadline has elapsed"
         );
     }
+
+    #[actix_web::test]
+    async fn invalid_tx_hash_encoding_test() {
+        let app = test::init_service(new_app()).await;
+
+        let invalid_payload = json!({});
```
Comment (vg-27) [inline src/tests.rs:420]: use valid json body, to make sure test fails on invalid tx_hash.

File: Cargo.toml [hunk 3]
```diff
@@ -28,3 +28,6 @@ clap = { version = "4.4.7", features = ["derive"] }
 thiserror = "1.0.50"
 hex = "0.4.3"
 rand = "0.8.5"
+data-encoding = "2.5.0"
```
Comment (vg-27) [inline Cargo.toml:31]: it would be helpful if dependencies are sorted alphabetically.

File: src/tests.rs (hunk 2 above)
Comment (roshanr95) [reply src/tests.rs:420]: there's no valid body for an invalid hash anyway since there's no code to interpret it

File: src/cgroups.rs [hunk 4]
```diff
@@ -55,7 +54,7 @@ impl Cgroups {
 }
 
 fn get_cgroups() -> Result<Vec<String>, std::io::Error> {
-    Ok(fs::read_dir("/sys/fs/cgroup")?
+    Ok(fs::read_dir("/sys/fs/cgroup/memory")?
```
Comment (vg-27) [inline src/cgroups.rs:57]: can you double-check on this? cgroup_v2 doesn't have `memory` dir on my system.

Comment (roshanr95) [reply src/cgroups.rs:57]: it's for cgroup v1 since the enclave kernel has v1, not v2

File: Cargo.toml (hunk 3 above)
Comment (roshanr95) [reply Cargo.toml]: done

**Comment by roshanr95** [comment]
@vg-27 all good to go?

File: README.md [hunk 5]
```diff
@@ -11,73 +11,38 @@ Oyster Serverless is a cutting-edge, high-performance serverless computing platf
...
-
-</br>
-
 ## Running serverless application
 
 <b>Run the serverless application :</b>
 
 ```
-cargo build --release && sudo ./target/x86_64-unknown-linux-musl/release/serverlessrust
+cargo build --release --target x86_64-unknown-linux-musl && sudo ./target/x86_64-unknown-linux-musl/release/serverlessrust
```
Comment (vg-27) [inline README.md:24]: update the `--signer` option.

**Review by vg-27 — APPROVED** [review]

**Pull request merged**